tools/server-admin/src/main/java/org/projectnessie/tools/admin/cli/CutHistory.java

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.admin.cli;
+
+import static org.projectnessie.versioned.storage.cleanup.Cleanup.createCleanup;
+import static org.projectnessie.versioned.storage.common.logic.Logics.repositoryLogic;
+import static org.projectnessie.versioned.storage.versionstore.TypeMapping.hashToObjId;
+
+import java.time.Duration;
+import java.time.Instant;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.storage.cleanup.CleanupParams;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "cut-history",
+    mixinStandardHelpOptions = true,
+    description = {
+      "Advanced commit log manipulation command that removes parents from the specified commit. "
+          + "Use with extreme caution. This command will make Nessie caches inconsistent with persisted data, "
+          + "therefore, it is preferable to run this command when Nessie servers are shut down. At the very least, "
+          + "all Nessie servers should be restarted as soon as possible after this command completes. "
+          + "Subsequently, it may be worth running the `cleanup-repository` command."
+    })
+public class CutHistory extends BaseCommand {
+  @CommandLine.Option(
+      names = {"-c", "--commit"},
+      required = true,
+      description = "The hash of the commit that gets detached from its parents.")
+  private String cutPoint;
+
+  @CommandLine.Option(
+      names = {"-R", "--retry"},
+      defaultValue = "0",
+      description = "Number of retries for idempotent sub-operations.")
+  private int numRetries;
+
+  @CommandLine.Option(
+      names = {"--dry-run"},
+      description = "Perform all operations, but do not modify any objects.")
+  private boolean dryRun;
+
+  @Override
+  public Integer call() throws Exception {
+    if (!repositoryLogic(persist).repositoryExists()) {
+      spec.commandLine().getErr().println("Nessie repository does not exist");
+      return EXIT_CODE_REPO_DOES_NOT_EXIST;
+    }
+
+    CleanupParams cleanupParams = CleanupParams.builder().dryRun(dryRun).build();
+
+    var cleanup = createCleanup(cleanupParams);
+    var params = cleanup.buildCutHistoryParams(persist, hashToObjId(Hash.of(cutPoint)));
+    var cutHistory = cleanup.createCutHistory(params);
+
+    var start = Instant.now();
+    var identifyResult = cutHistory.identifyAffectedCommits();
+    spec.commandLine()
+        .getOut()
+        .printf("Identified %d related commits.%n", identifyResult.affectedCommitIds().size());
+
+    int exitCode = 0;
+    for (int r = 0; r <= numRetries; r++) {
+      var result = cutHistory.cutHistory(identifyResult);
+
+      spec.commandLine()
+          .getOut()
+          .printf("Rewrote %d commits.%n", result.rewrittenCommitIds().size());
```

Comment:
Maybe adjust output for dry run - to not let users panic...

Reply:
Updated to not print this when there are no changes.

By the way, `result.rewrittenCommitIds()` is empty in dry runs.